src/Middleware/Ui.php

```diff
@@ -55,6 +63,11 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
             $session = $GLOBALS['injector']->getInstance(\Horde_Session::class);
             $registry = $GLOBALS['injector']->getInstance(\Horde_Registry::class);
 
+            // getting node information of Horde top-menubar: this will be saved in jsGlobalsHorde
+            $menu = $this->topbar->create('Passwd_Tree_Nodesobject', array('nosession' => true));
+            $menu =  $menu->getTree();
```

Comment:
This is not needed, right?

Reply:
I did not manage  to instantiate "Passwd_Tree_Nodesobject" directly. If I do, I get an error message saying that "Class UI.php was not found."

Also: the "topbar" class is used to create the tree object. The topbar class is a horde class, which creates a tree based on information about all the apps registered in Horde. I created the "Passwd_Tree_Nodesobject" class to filter the tree object made by the topbar, to get a more compact array with only the needed information (for the react app).

Reply:
I was just talking about the line `$menu =  $menu->getTree();`

Reply:
ok yes, so the getTree() method has to be called because it builds the initial tree (the Horde_Core_Topbar class does this) and the getNodes() only recreates/gets (not sure) the nodes (it leaves all the other info of the topbar behid...).
In general: when I do not use getTree() I also get an error saying UI.php was not found ;)